### ISSUE TITLE
need afterSelectionChange event for correct chart.getSelectedPoints

### ISSUE THREAD

**Issue by adumesny** [highcharts/highcharts#10839]
#### Expected behaviour
`chart.getSelectedPoints()` needs to be used when selection changes - maybe there needs to be a new **afterSelectionChanged** type of events to notify users when it is safe to call this method.

#### Actual behaviour

`getSelectedPoints()` it is off by one in `select` / `unselect` event and require most people to write a bunch of code to add/remove and manage that list in those basic event handlers... guessing sweep select would call each a bunch as well - people want to have a final selection changed event, not just a granular event.

doing a `setTimeout` hack on the `click` event works, but that's assumes user driven and not API select changed driven.

That is very common task and counter intuitive. see #9099 and countless others.

#### Product version
7.1.1

**Comment by KacperMadej** [comment]
Hi @adumesny 

Thank you for reporting about the problem.

There's a undocumented `afterSetState` for point that could be used instead of the setTimeout.
If this would work for you then we could label this issues as documentation enhancement and later add the missing docs to close the issue. What do you think about this option?

Here's a simple demo: http://jsfiddle.net/BlackLabel/vjuc7z6m/

Internal note: related to or duplicate of #9099

**Comment by adumesny** [comment]
@KacperMadej `afterSetState` is not a good solution because it's too generic/granular - only gets called after **each** selection item (not at the end) and no 'unselect' event (state of null apparently)

you can try my slightly modified version http://jsfiddle.net/adumesny/uhrdksn6/2/ 
ctrl+click a few points (all ok) then single click a fourth point, previous 3 get deselected but `afterSetState` still shows 4 items as unselect have not been processed by then.

It's also the wrong name - people expect select/selection to be part of the name.

Again people shouldn't have to write a unch of code to work around **basic** feature of: **tell me when selection changed** (select and de-select - the final state NOT individual ones you already support)

**Comment by KacperMadej** [comment]
This is because unselect runs after selection.

> It's also the wrong name - people expect select/selection to be part of the name.

`afterSetState` is a wrong name? Could you specify who are the people you are talking about? I mean, is this based on another tool, environment or an opinion? We value all and every reason, we will proceed this ticket with care - I just would like to know more about the case.

In general the situation is described by Paweł's comment in the related issue: https://github.com/highcharts/highcharts/issues/9099#issuecomment-426208037
In short: it needs to work like this, so you could prevent default behavior before it happens

> tell me when selection changed

This should be fairly easy enhancement to do - additional event could be fired after the select default function. For now please try and test a demo here: http://jsfiddle.net/BlackLabel/5raf7Loe/
The code for a proper enhancement will be slightly different, but general flow will remain. Please let me know is this what you are looking for or should I adjust the code further.

**Comment by adumesny** [comment]
> It's also the wrong name - people expect select/selection to be part of the name.
> `afterSetState` is a wrong name?
what I meant is if someone is looking for "tell me when selection has changed" this might not be the right name. It's perfectly fine for the current implementation (all internal/itermediate state changes including hover, etc...)

that's why I'm suggesting a new event. 

Your demo afterSelectionChanged does what I was looking for, the only thing is it shouldn't trigger if selection hasn't actually changed (say a select/unselect event returns false and doesn't change selection - see http://jsfiddle.net/adumesny/2zw0k9L3/10/) but pretty close!

Thanks

**Comment by KacperMadej** [comment]
@adumesny Thank you for all the info and the testing of the POC.
Here's a demo with the event moved slightly, so it will be fired only after the default select event will run: http://jsfiddle.net/BlackLabel/vdxarz90/

If the functionality should be changed or improved in any way please let me know and I will prepare a better demo.

**Comment by adumesny** [comment]
works perfectly now. thank you!
